sample_metadata/parser/generic_parser.py

```diff
@@ -148,21 +207,21 @@ def file_contents(self, filename) -> Optional[str]:
         with open(filename, encoding='utf-8') as f:
             return f.read()
 
-    def file_exists(self, filename: str) -> bool:
+    async def file_exists(self, filename: str) -> bool:
         """Determines whether a file exists"""
         path = self.file_path(filename)
 
         if path.startswith('gs://'):
-            blob = self.get_blob(filename)
+            blob = await self.get_blob(filename)
```

Comment:
Do I understand correctly that `get_blob` is the slowest part of checking the object existence? So file_exists won't get faster after this PR because `get_blob` is done synchronously?

Reply:
It's probably the slowest call, but making it async (even though it's technically not) means we can run them in parallel using asyncio without any extra effort. Or we can make the function async, and then we can run the function in parallel.